.travis.yml

```diff
@@ -91,160 +53,150 @@ matrix:
-          stage: Cron tests
-          env: NUMPY_VERSION=dev MATPLOTLIB_VERSION=dev EVENT_TYPE='cron'
-               CONDA_DEPENDENCIES=''
-               PIP_DEPENDENCIES=$DEV_PIP_DEP
+        - language: python
+          python: 3.7
+          stage: Final tests
+          env: TOXENV="test-devdeps"
+               EVENT_TYPE='cron'
 
         # Run documentation link check in a cron job.
         # Was originally in CircleCI doc build but links are too flaky, so
         # we moved it here instead.
-        # NOTE: matplotlib pinning should be consistent with CircleCI HTML job.
-        - os: linux
+        - language: python
+          python: 3.7
           stage: Cron tests
-          env: SETUP_CMD='build_docs -b linkcheck' EVENT_TYPE='cron'
-               MATPLOTLIB_VERSION="<3.1"
-               PIP_DEPENDENCIES="" CONDA_DEPENDENCIES=""
-               INSTALL_WITH_PIP=True
-               EXTRAS_INSTALL="docs"
-
-        # Testing with Travis provided Python3.8. Change this job to a regular
-        # one once conda supports python3.8. Make sure we use released numpy rather
-        # than some kind of dev coming natively from the Travis environment
+          env: TOXENV='linkcheck'
+               EVENT_TYPE='cron'
+          addons:
+              apt:
+                  packages:
+                      - graphviz
+
+        # Testing with Travis provided Python3.8.
         - language: python
           stage: Final tests
           dist: xenial
-          python: 3.8-dev
-          env: INSTALL_WITH_PIP=True EXTRAS_INSTALL="test" USE_CI_HELPERS=False
-               OTHER_DEPENDENCY_PREFERENCES='numpy==1.17.3' PIP_FLAGS="--pre"
+          python: 3.8
 
     allow_failures:
-      - os: linux
-        stage: Final tests
-        env: PYTHON_VERSION=3.7 NUMPY_VERSION=dev SETUP_CMD='test --remote-data'
-             CONDA_DEPENDENCIES=''
-             PIP_DEPENDENCIES=$DEV_PIP_DEP
-             MATPLOTLIB_VERSION=dev
+        - language: python
+          python: 3.7
+          stage: Final tests
+          env: TOXENV="test-devdeps"
+               TOXPOSARGS="--remote-data"
 
 before_install:
 
+    # We need a full clone to make sure setuptools_scm works properly
+    - git fetch --unshallow .
+    - git fetch --depth=1000000
+
     # We need to use CCOMPILER otherwise Travis overwrites CC if we define it
     # in env: above.
     - if [ ! -z $CCOMPILER ]; then
         export CC=$CCOMPILER;
       fi
 
      # Check CC variable
-    - echo $CC
+    - echo "CC="$CC
 
     # Write configuration items to standard location to make sure they are
     # ignored (the tests will fail if not)
     - mkdir -p $HOME/.astropy/config/
     - printf "unicode_output = True\nmax_width = 500" > $HOME/.astropy/config/astropy.cfg
 
+    - if [[ $COVERAGE == True ]]; then
+        export TOXPOSARGS=$TOXPOSARGS" --cov-report=xml:"$TRAVIS_BUILD_DIR"/coverage.xml";
+      fi
+
+    - echo "TOXENG="$TOXENG
+    - echo "TOXARGS="$TOXARGS
```

Comment:
`TOXARGS` are not used anywhere, so I would remove it (arguably the point is to define all those in an environment anyway; but in any case, we shouldn't introduce anything just because it might be handy).

Reply:
For debugging it could be handy to have `TOXARGS='-v'` by default ?

Reply:
I agree with @saimn that ``-v`` would be good to have. We do also need to make sure we use a standardized name here once we use https://github.com/astropy/ci-helpers/pull/429.

Reply:
OK, if we use it for `-v` then it is fine to introduce.

Reply:
Done